### ISSUE TITLE
UNKNOWN: NoCredentialProviders if AWS_PROFILE is specified

### ISSUE THREAD

**Issue by noliva** [aws-observability/aws-otel-collector#156]
Hello,

I tried to set up locally the collector with the following configuration:
```yaml
version: "3"
services:
  aws-ot-collector:
    image: amazon/aws-otel-collector:latest
    command: ["--config=/etc/otel-agent-config.yaml", "--log-level=DEBUG"]
    environment:
      - AWS_PROFILE=p3
      - AWS_REGION=eu-west-1
    volumes:
      - ./config-test.yaml:/etc/otel-agent-config.yaml
      - ~/.aws:/root/.aws
    ports:
      - "55680:55680" # OTLP receiver
```

But when submitting traces I get `{"stack":"Error: 2 UNKNOWN: NoCredentialProviders: no valid providers in chain. Deprecated.\n\tFor verbose messaging see aws.Config.CredentialsChainVerboseErrors\n    at Object.exports.createStatusError`

But if instead of using `AWS_PROFILE` I add `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY` then it works, so my docker-compose.yaml file looks like:

```yaml
version: "3"
services:
  aws-ot-collector:
    image: amazon/aws-otel-collector:latest
    command: ["--config=/etc/otel-agent-config.yaml", "--log-level=DEBUG"]
    environment:
      - AWS_REGION=eu-west-1
      - AWS_ACCESS_KEY_ID=key
      - AWS_SECRET_ACCESS_KEY=secret
    volumes:
      - ./config-test.yaml:/etc/otel-agent-config.yaml
      - ~/.aws:/root/.aws
    ports:
      - "55680:55680" # OTLP receiver
```

My credentials file in `~/.aws/credentials` looks like
```
[p3]
aws_access_key_id=key
aws_secret_access_key=secret
```

And my `tracer.js` looks like:

```javascript
const { NodeTracerProvider } = require('@opentelemetry/node');
const { ConsoleSpanExporter, SimpleSpanProcessor } = require('@opentelemetry/tracing');
// const { CollectorTraceExporter } =  require('@opentelemetry/exporter-collector');
const { CollectorTraceExporter } = require('@opentelemetry/exporter-collector-grpc');
const { propagation } = require("@opentelemetry/api");

const { AwsXRayIdGenerator } = require('@aws/otel-aws-xray-id-generator');
const { AWSXRayPropagator } = require('@aws/otel-aws-xray-propagator');

propagation.setGlobalPropagator(new AWSXRayPropagator());

const provider = new NodeTracerProvider({
    idGenerator: new AwsXRayIdGenerator()
});

const otlpExporter = new CollectorTraceExporter({
    serviceName: 'service-hello',
    url: "localhost:55680"
});
const otlpProcessor = new SimpleSpanProcessor(otlpExporter);
provider.addSpanProcessor(otlpProcessor);

provider.register();
```

So I'm not really sure why this is not working

**Comment by mxiamxia** [comment]
Hi @noliva , I can't reproduce the issue with creating different AWS profiles and it all can be recognized in our testing.  Can you double check your `p3` profile has the appropriate X-Ray permission for publishing data? See X-Ray/Collector permission page [here](https://aws-otel.github.io/docs/setup/permissions).

```
- AWS_PROFILE=p3
- AWS_REGION=eu-west-1
```

**Comment by noliva** [comment]
@mxiamxia thanks for the response.

I've checked and if I provide the credentials like 

```
- AWS_ACCESS_KEY_ID=key
- AWS_SECRET_ACCESS_KEY=secret
```

then it all works, and those credentials are the same one in my p3 profile

**Comment by praetp** [comment]
Any update on this?

**Comment by jefchien** [comment]
Hi @noliva and @praetp, is this still an issue? I haven't been able to replicate it.

I removed all the profiles in `~/.aws/credentials` aside from the profile I wanted to use. I used `examples/docker/docker-compose.yaml` with the `AWS_PROFILE` environment variable set under `aws-ot-collector` and `ot-sample-app` and had no problems. Had some issues without the `ot-sample-app` `AWS_PROFILE` set since my AWS config no longer had a default profile, but that was a AWS Java SDK exception rather than the AWS Go SDK error seen above.
